protocol/vm/context.go

```diff
@@ -0,0 +1,31 @@
+package vm
+
+// Context contains the execution context for the virtual machine.
+//
+// By convention, variables of this type have the name context, _not_
+// ctx (to avoid confusion with context.Context).
+type Context struct {
+	VMVersion uint64
+	Code      []byte
+	Arguments [][]byte
+
+	TxVersion *uint64
+
+	BlockHash            *[]byte
```

Comment:
It would be nice to document how the optional fields work, maybe in the godoc for type Context or func Verify. Something brief would be fine, e.g. something like what I wrote in an earlier comment:

```
// Fields below this point are optional. If the corresponding opcode
// executes on a nil field, Verify will return ErrContext.
```